### PR TITLE
[2.10] Updates to the deferrable operators guide

### PR DESCRIPTION
**Pull request by TJaniF**
Adjusted the language to be correct with https://github.com/apache/airflow/pull/38674 implemented and added a code example on how to implement direct triggering.

Side note: I'll update all the main docs links once the release is out :)

### CODE REVIEW

File: learn/deferrable-operators.md [hunk 1]
```diff
@@ -45,7 +46,7 @@ With traditional operators, a task submits a job to an external system such as a
...
+With deferrable operators, worker slots are released when a task is polling for the job status. When the task is deferred, the polling process is offloaded as a trigger to the triggerer, and the worker slot becomes available. Note that in Airflow 2.10+ some deferrable operators directly enter a deferred state without going to a worker first, see [Triggering Deferral from Start](http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow/stable/authoring-and-scheduling/deferring.html#triggering-deferral-from-task-start). The triggerer can run many asynchronous polling tasks concurrently, and this prevents polling tasks from occupying your worker resources. When the terminal status for the job is received, the operator resumes the task, taking up a worker slot while it finishes. The following image illustrates the process:
```
Comment (kentdanas) [inline learn/deferrable-operators.md:49]:
```suggestion
With deferrable operators, worker slots are released when a task is polling for the job status. When the task is deferred, the polling process is offloaded as a trigger to the triggerer, and the worker slot becomes available. Note that in Airflow 2.10+, some deferrable operators directly enter a deferred state without going to a worker first, see [Triggering Deferral from Start](http://apache-airflow-docs.s3-website.eu-central-1.amazonaws.com/docs/apache-airflow/stable/authoring-and-scheduling/deferring.html#triggering-deferral-from-task-start). The triggerer can run many asynchronous polling tasks concurrently, and this prevents polling tasks from occupying your worker resources. When the terminal status for the job is received, the operator resumes the task, taking up a worker slot while it finishes. The following image illustrates the process:
```
I wonder if it makes sense to pull this out into a `note` box? Do you think the average Airflow user would need to know this up front? Or can it be more of an aside that might only be relevant to certain users?

File: learn/deferrable-operators.md [hunk 2]
```diff
@@ -136,7 +137,7 @@ Triggers are designed to be highly available. You can implement this by starting
 
 If you have an operator that would benefit from being asynchronous, but does not exist in OSS Airflow yet, you can create your own by writing a deferrable operator and trigger class. You can also defer a task several times if needed.
 
-Get a template for a custom deferrable operator and custom trigger class, by clicking the dropdown below:
+Get a template for a custom deferrable operator and custom trigger class, by clicking the dropdown below. Make sure to adjust the classpath for your trigger's `.serialize` method (currently `include.deferrable_operator_template.MyTrigger`) to your file structure.
```
Comment (kentdanas) [inline learn/deferrable-operators.md:140]:
```suggestion
Get a template for a custom deferrable operator and custom trigger class, by clicking the dropdown below. Make sure to adjust the classpath for your trigger's `.serialize` method (currently `include.deferrable_operator_template.MyTrigger`) to match your file structure.
```
If this isn't the right update let me know, I was a bit confused by the original sentence.

File: learn/deferrable-operators.md [hunk 3]
```diff
@@ -147,4 +148,15 @@ Get a template for a custom deferrable operator and custom trigger class, by cli
 
 Note that when developing a custom trigger, you need to restart your triggerer to pick up any changes you make, since the triggerer caches the trigger classes. Additionally, all information you pass between the triggerer and the worker must be JSON serializable.
 
-See [Writing Deferrable Operators](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/deferring.html#writing-deferrable-operators) for more information.
+See [Writing Deferrable Operators](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/deferring.html#writing-deferrable-operators) for more information. 
+
+Airflow 2.10 includes options to implement direct deferral without ever being picked up by a worker. See the code below for a deferrable operator circumventing the `.execute()` method. When using this template, make sure to adjust the classpath for your trigger (currently `include.deferrable_operator_template.MyTrigger`) in both, the `.serialize` method, as well as the `StartTriggerArgs`  to your file structure.
```
Comment (kentdanas) [inline learn/deferrable-operators.md:153]:
```suggestion
Airflow 2.10 includes the options to implement direct deferral without ever being picked up by a worker. See the code below for a deferrable operator circumventing the `.execute()` method. When using this template, make sure to adjust the classpath for your trigger (currently `include.deferrable_operator_template.MyTrigger`) in both the `.serialize` method and the `StartTriggerArgs`  to match your file structure.
```

**Review by kentdanas — APPROVED** [review]
A few minor comments. If my updates make sense then I don't need to look again, but feel free to tag me again if you want a re-review on anything.

File: learn/deferrable-operators.md (hunk 1 above)
Comment (TJaniF) [reply learn/deferrable-operators.md]: Good point, and no, this is only relevant to users who write their own deferrable operators really, will move it to a note :)

**Review by TJaniF — COMMENTED** [review]
Thank you @kentdanas ! Made all the changes :)

**Pull request merged**